src/firewall/core/nftables.py

```diff
@@ -239,10 +239,26 @@ def __init__(self, fw):
         self._nft_handle.set_json_output(True)
 
     def nft_cmd(self, json_root):
+        if isinstance(json_root, list):
+            # The caller gave us the JSON in a list, so we can drop the
+            # reference to the memory. Unpack the list, and delete the entry.
+            l = json_root
+            (json_root,) = l
+            del l[0]
         json_root_str = json.dumps(json_root)
+        del json_root
```

Comment:
I'm not sure this does anything if other areas of the code, e.g. the caller, are holding a reference. It would have to be `json_root.clear()` to remove the dict entries.

Reply:
it would, if `json_root` argument were a list.

I added commit "improvement(nftables): free json data earlier during nft_cmd()" to do that.